src/components/Settings/PhotosFolder.vue

```diff
@@ -0,0 +1,111 @@
+<template>
+	<div class="folder">
+		<component :is="rootFolderIcon" v-if="path === '/'" />
+		<Folder v-else />
+		<span class="folder__info">
+			<div class="folder__path">{{ folderName }}</div>
+			<div v-if="subname !== ''">
+				{{ subname }}
+			</div>
+		</span>
+		<NcButton v-if="canDelete"
+			type="tertiary"
+			:aria-label="t('photos', 'Delete source directory')"
+			@click="emitRemoveSourceFolder">
+			<template #icon>
+				<Close :size="20" />
+			</template>
+		</NcButton>
+	</div>
+</template>
+
+<script>
+import { defineComponent } from 'vue'
+
+import Folder from 'vue-material-design-icons/Folder.vue'
+import Close from 'vue-material-design-icons/Close.vue'
+
+import { NcButton } from '@nextcloud/vue'
+import { translate as t } from '@nextcloud/l10n'
+
+export default defineComponent({
+	name: 'PhotosFolder',
+
+	components: {
+		NcButton,
+		Folder,
+		Close,
+	},
+
+	props: {
+		path: {
+			type: String,
+			required: true,
+		},
+		canDelete: {
+			type: Boolean,
+			default: false,
+		},
+		rootFolderLabel: {
+			type: String,
+			required: true,
+		},
+		rootFolderIcon: {
+			type: Object,
+			required: true,
+		},
+	},
+
+	emits: ['remove-folder'],
+
+	computed: {
+		folderName() {
+			if (this.path === '/') {
+				return this.rootFolderLabel
+			} else {
+				return this.path.split('/').pop()
+			}
+		},
+
+		subname() {
+			const slashesCount = (this.path.match(/\//g) ?? []).length
+
+			switch (slashesCount) {
+			case 1:
+				return ''
+			case 2:
+				return this.path.split('/').splice(0, 2).join('/')
+			default:
+				return this.path.split('/').splice(0, 3).join('/')
+			}
```

Comment:
What is this sorcery? :grin:

Reply:
To properly display the path of the folder. Let me add a comment.

Full path | Displayed path
-- | --
/ | Home
/a | nothing
/a/b | /a
/a/b/c | /a/b

Reply:
More comments the better :)